adapters/showheroes/showheroes.go

```diff
@@ -0,0 +1,223 @@
+	if err != nil {
+		return err
+	}
+
+	imp.BidFloor = eurFloor
+	imp.BidFloorCur = "EUR"
+
+	return nil
+}
+
+func getPrebidChannel(request *openrtb2.BidRequest) (string, string) {
+	var channelName string
+	var channelVersion string
+	reqExt := &openrtb_ext.ExtRequest{}
+
+	if err := jsonutil.Unmarshal(request.Ext, reqExt); err == nil && reqExt.Prebid.Channel != nil {
+		channelName = reqExt.Prebid.Channel.Name
+		channelVersion = reqExt.Prebid.Channel.Version
+	}
+	return channelName, channelVersion
+}
+
+func setPBSVersion(request *openrtb2.BidRequest, pbsVersion string) {
+	if pbsVersion == "" {
+		pbsVersion = version.VerUnknown
+	}
+	source := request.Source
+	if source == nil {
+		source = &openrtb2.Source{}
+		request.Source = source
+	}
+
+	sourceExtMap := make(map[string]json.RawMessage)
+	if source.Ext != nil {
+		if err := jsonutil.Unmarshal(source.Ext, &sourceExtMap); err != nil {
+			return
+		}
+	}
+	sourceExtMap["pbs"] = json.RawMessage(`{"pbsv":"` + pbsVersion + `", "pbsp": "go"}`)
+	if ext, err := json.Marshal(sourceExtMap); err == nil {
+		source.Ext = ext
+	}
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, extra *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if err := validate(request); err != nil {
+		return nil, []error{err}
+	}
+	var errors []error
+	validImps := make([]openrtb2.Imp, 0, len(request.Imp))
+
+	prebidChannelName, channelVersion := getPrebidChannel(request)
+
+	for _, imp := range request.Imp {
+		if err := processImp(&imp, extra); err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		if imp.DisplayManager == "" {
+			imp.DisplayManager = prebidChannelName
+			imp.DisplayManagerVer = channelVersion
+		}
+		validImps = append(validImps, imp)
+	}
+
+	if len(validImps) == 0 {
+		return nil, errors
+	}
+
+	setPBSVersion(request, version.Ver)
+
+	request.Imp = validImps
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	return []*adapters.RequestData{
+		{
+			Method:  "POST",
+			Uri:     a.endpoint,
+			Body:    reqJSON,
+			Headers: headers,
+			ImpIDs:  openrtb_ext.GetImpIDs(request.Imp),
+		},
+	}, errors
+}
+
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{fmt.Errorf("unexpected status code: %d", response.StatusCode)}
+	}
+	var bidResponse openrtb2.BidResponse
+	if err := jsonutil.Unmarshal(response.Body, &bidResponse); err != nil {
+		return nil, []error{err}
+	}
+
+	bidderResponse := adapters.NewBidderResponse()
+	bidderResponse.Currency = bidResponse.Cur
```

Comment:
Should we check `bidResponse.Cur` before assigning? `NewBidderResponse()` calls `NewBidderResponseWithBidsCapacity()`, a function that returns a `BidderResponse` wuth the `Currency` field set to `"USD"`. If `bidResponse.Cur` is an empty string, we'd overwrite `bidderResponse.Currency`'s default value of `"USD"`. Do we want that?

Reply:
I leave it up to you. If you are ok with the current behavior, I'm fine with it. Let me know

Reply:
I'm fine with this. We are always providing currency in the bid response.